quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusTextDocumentService.java

```diff
@@ -218,6 +226,37 @@ public void didSave(DidSaveTextDocumentParams params) {
 		});
 	}
 
+	@Override
+	public CompletableFuture<List<Either<Command, CodeAction>>> codeAction(CodeActionParams params) {
+		QuarkusProjectInfoParams projectInfoParams = createProjectInfoParams(params.getTextDocument(), null);
+		return getProjectInfoCache().getQuarkusProjectInfo(projectInfoParams).thenComposeAsync(projectInfo -> {
+			if (projectInfo.getProperties().isEmpty()) {
+				return null;
+			}
+			// then get the Properties model document
+			return getPropertiesModel(params.getTextDocument(), (cancelChecker, document) -> {
+				String uri = params.getTextDocument().getUri();
+				return getQuarkusLanguageService()
+						.doCodeActions(params.getContext(), params.getRange(), document, projectInfo,
+								sharedSettings.getFormattingSettings()) //
+						.stream() //
+						.map(ca -> {
+							if (codeActionLiteralSupport) {
+								Either<Command, CodeAction> e = Either.forRight(ca);
+								return e;
+							} else {
+								List<Object> arguments = Arrays.asList(uri, document.getDocument().getVersion(),
+										ca.getEdit().getDocumentChanges().get(0).getLeft().getEdits());
+								Command command = new Command(ca.getTitle(), "_quarkus.applyCodeAction", arguments);
```

Comment:
do we really still need client specific implementations to apply code actions?

Reply:
It's a copy paste from LSP4XML code action. If you wish I can remove it.

Reply:
@fbricon do you want i remove this code?

Reply:
yup